apps/microtvm/arduino/template_project/tests/test_arduino_microtvm_api_server.py

```diff
@@ -0,0 +1,100 @@
+from unittest import mock
+from pathlib import Path
+import sys
+
+import pytest
+
+sys.path.insert(0, str(Path(__file__).parent.parent))
+import microtvm_api_server
+
+sys.path.pop(0)
+
+
+class TestGenerateProject:
+    DEFAULT_OPTIONS = {"arduino_cli_cmd": "arduino-cli", "arduino_board": "nano33ble"}
+
+    def test_print_c_array(self):
+        handler = microtvm_api_server.Handler()
+        c_arr = handler._print_c_array([1, 32, 32, 3])
+        assert c_arr == "{1, 32, 32, 3}"
+
+    def _set_pathlib_path_exists(self, value):
+        with mock.patch.object(Path, "exists") as mock_exists:
+            mock_exists.return_value = value
+
+    @mock.patch("pathlib.Path")
+    def test_find_modified_include_path(self, mock_pathlib_path):
+        handler = microtvm_api_server.Handler()
+
+        project_dir = mock_pathlib_path("/dummy/project")
+        file_path = (
+            project_dir
+            / "src"
+            / "standalone_crt"
+            / "src"
+            / "runtime"
+            / "crt"
+            / "graph_executor"
+            / "load_json.c"
+        )
+
+        # Should return C standard libs unmodified
+        clib_output = handler._find_modified_include_path(project_dir, file_path, "math.h")
+        assert clib_output == "math.h"
+
+        # If import already works, should return unmodified
+        valid_arduino_import = "../../../../include/tvm/runtime/crt/platform.h"
+        self._set_pathlib_path_exists(True)
+        valid_output = handler._find_modified_include_path(
+            project_dir, file_path, valid_arduino_import
+        )
+        assert valid_output == valid_arduino_import
+
+    BOARD_CONNECTED_OUTPUT = bytes(
+        "Port         Type              Board Name          FQBN                        Core             \n"
+        "/dev/ttyACM0 Serial Port (USB) Arduino Nano 33 BLE arduino:mbed_nano:nano33ble arduino:mbed_nano\n"
```

Comment:
can you include an adversarial example with fqbn == "arduino:mbed_nano:nano33" and prove you can select it?